### PR TITLE
feat(wallet): Accounts Tab V2

### PR DESCRIPTION
**Pull request by Douglashdaniel**
## Description 
- Updates the `Accounts tab` to the new `Wallet V2` designs

Resolves <https://github.com/brave/brave-browser/issues/31432>

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
1. Go to the `Accounts` tab and make sure everything works correctly.

https://github.com/brave/brave-core/assets/40611140/5bebc13b-975f-44c4-bc79-648b6294053e

Screenshot 47

Screenshot 48

### CODE REVIEW

File: components/brave_wallet_ui/components/desktop/views/accounts/accounts.tsx [hunk 1]
```diff
@@ -54,21 +54,12 @@ export const Accounts = () => {
...
   // memos
-  const primaryAccounts = React.useMemo(() => {
+  const derivedAccounts = React.useMemo(() => {
     return accounts.filter((account) => getAccountType(account) === 'Primary')
   }, [accounts])
 
-  const secondaryAccounts = React.useMemo(() => {
+  const importedAccounts = React.useMemo(() => {
     return accounts.filter((account) => getAccountType(account) === 'Secondary')
   }, [accounts])
```
Comment (supermassive) [inline components/brave_wallet_ui/components/desktop/views/accounts/accounts.tsx:64]: It seems filtering should be now on 
`account.accountId.kind === BraveWallet.AccountKind.kDerived`
`account.accountId.kind === BraveWallet.AccountKind.kImported`

**Review by supermassive — APPROVED** [review]
wallet-core LGTM

File: components/brave_wallet_ui/components/desktop/views/accounts/accounts.tsx (hunk 1 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/views/accounts/accounts.tsx]: Works for me! Updated!

File: components/brave_wallet_ui/components/desktop/views/accounts/accounts.tsx [hunk 2]
```diff
@@ -84,92 +80,117 @@ export const Accounts = () => {
...
+      </Column>
+
+      {importedAccounts.length !== 0 &&
+        <>
+          <Row
+            padding='8px'
+            justifyContent='flex-start'
+          >
+            <SectionTitle
+            >
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/desktop/views/accounts/accounts.tsx:119]: nit: closing bracket
```suggestion
            <SectionTitle>
```

File: components/brave_wallet_ui/components/desktop/views/accounts/accounts.tsx [hunk 3]
```diff
@@ -84,92 +80,117 @@ export const Accounts = () => {
...
+                account={account}
+              />
+            )}
+          </Column>
+        </>
+      }
+
+      {
+        (Object.keys(trezorAccounts).length !== 0 ||
+          Object.keys(ledgerAccounts).length !== 0) &&
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/desktop/views/accounts/accounts.tsx:140]: perf: calculate keys once and save to a variable

File: components/brave_wallet_ui/components/desktop/views/accounts/accounts.tsx [hunk 4]
```diff
@@ -84,92 +80,117 @@ export const Accounts = () => {
...
+        <>
+          <Row
+            padding='8px'
+            justifyContent='flex-start'
+          >
+            <SectionTitle
+            >
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/desktop/views/accounts/accounts.tsx:147]: nit: closing bracket placement

File: components/brave_wallet_ui/components/shared/icon-stacks/icon-stacks.style.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+import styled from 'styled-components'
+import * as leo from '@brave/leo/tokens/css'
+import {
+  AssetIconProps,
+  AssetIconFactory
+} from '../style'
+
+export const StackContainer = styled.div<{
+  width: number
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/shared/icon-stacks/icon-stacks.style.ts:14]: let's use a string here so we can use values other than px

File: components/brave_wallet_ui/components/desktop/views/accounts/accounts.tsx (hunk 2 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/views/accounts/accounts.tsx]: Fixed!

File: components/brave_wallet_ui/components/shared/icon-stacks/token-icons-stack.tsx [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+    tokens
+  } = props
+
+  // queries
+  const { data: networks = [] } = useGetVisibleNetworksQuery()
+
+  // Memos
+  const additionalTokensLength = React.useMemo(() => {
+    return tokens.length - 3
+  }, [tokens])
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/shared/icon-stacks/token-icons-stack.tsx:55]: perf: not compute-heavy enough to warrant memoization

File: components/brave_wallet_ui/components/shared/icon-stacks/token-icons-stack.tsx [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+  ])
+
+  // Methods
+  const calculateIconLeftPosition =
+    React.useCallback((index: number) => {
+      if (index === 0) {
+        return 4
+      }
+      return (index * 12) + 4
+    }, [])
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/shared/icon-stacks/token-icons-stack.tsx:84]: perf: can be defined as a function outside of the component

File: components/brave_wallet_ui/components/desktop/views/accounts/accounts.tsx [hunk 8]
```diff
@@ -84,92 +80,116 @@ export const Accounts = () => {
...
+                {sortAccountsByName(ledgerAccounts[key])
+                  .map((account: WalletAccountType) =>
+                    <AccountListItem
+                      key={account.accountId.uniqueKey}
+                      onClick={onSelectAccount}
+                      account={account}
+                    />
+                  )}
+              </Column>
+            )}
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/desktop/views/accounts/accounts.tsx:188]: can we memoize this?

File: components/brave_wallet_ui/components/desktop/views/accounts/accounts.tsx [hunk 9]
```diff
@@ -84,92 +80,116 @@ export const Accounts = () => {
...
+                {sortAccountsByName(trezorAccounts[key])
+                  .map((account: WalletAccountType) =>
+                    <AccountListItem
+                      key={account.accountId.uniqueKey}
+                      onClick={onSelectAccount}
+                      account={account}
+                    />
+                  )}
+              </Column>
+            )}
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/desktop/views/accounts/accounts.tsx:171]: can we memoize this?

Comment (josheleonard) [inline components/brave_wallet_ui/components/desktop/views/accounts/accounts.tsx:171]: can we memoize this?

**Review by josheleonard — CHANGES_REQUESTED** [review]

**Review by josheleonard — APPROVED** [review]
++

**Review by mkarolin — APPROVED** [review]
`strings`++

**Pull request merged**